src/Renci.SshNet/SshNetLoggingConfiguration.cs

```diff
@@ -0,0 +1,26 @@
+﻿#nullable enable
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using Renci.SshNet.Common;
+
+namespace Renci.SshNet
+{
+    /// <summary>
+    /// Allows configuring the logging for internal logs of SSH.NET.
+    /// </summary>
+    public static class SshNetLoggingConfiguration
+    {
+        internal static ILoggerFactory LoggerFactory { get; private set; } = NullLoggerFactory.Instance;
+
+        /// <summary>
+        /// Initializes the logging for SSH.NET.
+        /// </summary>
+        /// <param name="loggerFactory">The logger factory.</param>
+        public static void InitializeLogging(ILoggerFactory loggerFactory)
+        {
+            ThrowHelper.ThrowIfNull(loggerFactory);
+            LoggerFactory = loggerFactory;
+        }
```

Comment:
Maybe allow null here as a way to turn it off?

```suggestion
        /// <param name="loggerFactory">
        /// The logger factory, or <see langword="null"/> to disable logging.
        /// </param>
        public static void InitializeLogging(ILoggerFactory? loggerFactory)
        {
            LoggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
        }
```

Reply:
Imho that's what `NullLoggerFactory.Instance` is already there for. Also the usual way to turn off logs for specific components would be to configure the log level for the category (`Renci.SshNet`) to `LogLevel.None`.